src/models/workout.js

```diff
@@ -125,27 +125,35 @@ const WorkoutModel = {
     }
   },
 
-  updateSetItems: async ({ workoutItemIdArray, workoutSetId }) => {
+  updateSetItems: async ({ workoutItemArray, userId }) => {
     try {
+      let { data: daysIdSet } = await supabase.from('workout_sets').select().match({ users_id: userId });
+
       let queryValForNew = [];
-      workoutItemIdArray.map(async (item) => {
-        if (item.id == null) {
-          let val = {
-            workout_items_id: item.workoutItemId,
-            workout_sets_id: Number(workoutSetId),
-            set_order: item.order,
-            reps: item.reps,
-            sets: item.sets,
-          };
-          queryValForNew.push(val);
-        } else {
-          let { error } = await supabase
-            .from('workout_set_items')
-            .update({ set_order: item.order, reps: item.reps, sets: item.sets })
-            .match({ id: item.id });
-
-          supabaseErrorCheck(error);
-        }
+      workoutItemArray.map(async (dayItems, i) => {
+        let filteredDayItems = dayItems.filter((setItem) => {
+          return setItem.workout_item !== null;
+        });
+
+        filteredDayItems.map(async (item, j) => {
+          if (item.id == null) {
```

Comment:
there are too many loops here.
remove the filteredDayItems and just map through the dayItems.
```suggestion
          if (item.id === null && item.wokrout_item !== null) {
```

I think this works.